socs/Lakeshore/Lakeshore372.py

```diff
@@ -1295,7 +1295,41 @@ def set_curve(self, _file):
             self._set_data_point(point[0], point[1], point[2])
 
         # refresh curve attributes
-        return self.get_header()
+        self.get_header()
+
+    def _check_curve(self, _file):
+        """After setting a data point for calibration curve,
+        use CRVPT? command from get_data_point() to check
+        that all points of calibration curve  were uploaded.
+        If not, re-upload points.
+
+        :param _file: calibration curve file
+        :type _file: str
+        """
+
+        with open(_file) as f:
+            content = f.readlines()
+
+        #skipping header info
+
+        values = []
+        for i in range(9, len(content)):
+            values.append(content[i].strip().split()) #data points that should have been uploaded
+
+
+        for j in range(1, 201):
+            try:
+                resp = self.get_data_point(j) #response from the 372
+                point = values[j-1]
+                units = float(resp[0])
+                temperature = float(resp[1])
+                assert units == float(point[1]), "Point number %s not uploaded"%point[0]
+                assert temperature == float(point[2]), "Point number %s not uploaded"%point[0]
+                print("Successfully uploaded %s, %s" %(units,temperature))
+            #if AssertionError, tell 372 to re-upload points
+            except:
```

Comment:
We're expecting the AssertionError here, so we can explicitly except it:
```
except AssertionError:
    code to reupload
```

This'll mean other errors won't try to reupload points, and should be handled separately if they arise.